test/join.jl

```diff
@@ -346,8 +343,8 @@ module TestJoin
         @test levels(join(B, A, on=:b, kind=:inner)[:b]) == ["a", "b", "c"]
         @test levels(join(A, B, on=:b, kind=:left)[:b]) == ["d", "c", "b", "a"]
         @test levels(join(B, A, on=:b, kind=:left)[:b]) == ["a", "b", "c"]
-        @test levels(join(A, B, on=:b, kind=:right)[:b]) == ["d", "c", "b", "a"]
-        @test levels(join(B, A, on=:b, kind=:right)[:b]) == ["a", "b", "d", "c"]
+        @test levels(join(A, B, on=:b, kind=:right)[:b]) == ["a", "b", "c"]
```

Comment:
Do these changes here and below to Categorical levels adhere to the ordering requests made in https://github.com/JuliaData/DataFrames.jl/issues/1257? I don't see why restricting the auto promotion of columns should affect these